### PR TITLE
Removed symfony BC layer for old symfony versions

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Removed
 - Removed BC layer for older symfony versions
```

### CODE REVIEW

File: Form/Type/FormatterType.php [hunk 1]
```diff
@@ -212,11 +215,6 @@ public function configureOptions(OptionsResolver $resolver)
             'choices' => $formatters,
         ];
 
-        // NEXT_MAJOR: Remove the method_exists hack when dropping support for symfony < 2.7
-        if (count($formatters) > 1 && method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
```
Comment (jordisala1991) [inline Form/Type/FormatterType.php]: The first part of the if is no longer needed? NEXT_MAJOR comment only talks about removing `method_exists`

File: Form/Type/FormatterType.php [hunk 2]
```diff
@@ -212,11 +215,6 @@ public function configureOptions(OptionsResolver $resolver)
             'choices' => $formatters,
```
Comment (jordisala1991) [inline Form/Type/FormatterType.php]: This should get moved to the `format_field_options`

File: Form/Type/FormatterType.php (hunk 1 above)
Comment (core23) [reply Form/Type/FormatterType.php]: Don't know why we should only add this option when we have at least one formatter.

Comment (jordisala1991) [reply Form/Type/FormatterType.php]: https://github.com/sonata-project/SonataFormatterBundle/pull/211#issuecomment-240460367

Here is the discussion, not sure if it applies now, but seems like when we have one formatter, is displayed as a hidden input, and on SF 2.8 atleast it does not have `choice_translation_domain`

Comment (core23) [reply Form/Type/FormatterType.php]: Fixed

**Comment by jordisala1991** [comment]
Triggering build again after test fixing.

**Comment by core23** [comment]
Please review again @jordisala1991

File: docs/reference/ckeditor_medias.rst [hunk 3]
```diff
@@ -84,7 +84,7 @@ Here is an example to alter `shortDescription` field of the `ProductAdmin`:
 
     // ...
 
-    $formMapper->add('shortDescription', 'sonata_formatter_type', array(
+    $formMapper->add('shortDescription', FormatterType::class, array(
```
Comment (greg0ire) [inline docs/reference/ckeditor_medias.rst]: Please add use statements

Comment (core23) [reply docs/reference/ckeditor_medias.rst]: Good catch!

**Pull request merged**